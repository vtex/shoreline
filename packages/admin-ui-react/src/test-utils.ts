import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
