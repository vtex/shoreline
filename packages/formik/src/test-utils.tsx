import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
