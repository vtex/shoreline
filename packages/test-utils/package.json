{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
