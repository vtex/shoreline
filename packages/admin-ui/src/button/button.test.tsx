import React from 'react'

import { render, axe } from '../test-utils'
import { Button } from './index'

describe('button', () => {
  it('should not have a11y violations', async () => {
    const { container } = render(<Button>Button</Button>)

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
