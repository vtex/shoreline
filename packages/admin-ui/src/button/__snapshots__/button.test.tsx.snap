// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should match snapshot 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #FFFFFF;
  background: hsla(222, 63%, 43%, 1);
  padding: 0.25rem 0.75rem;
  height: 2.25rem;
}

.emotion-2:hover {
  color: #FFFFFF;
  background: hsla(222, 63%, 34%, 1);
}

.emotion-2:active {
  color: #FFFFFF;
  background: hsla(222, 63%, 25%, 1);
}

.emotion-2:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

<button
    class="emotion-2"
    type="button"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-0"
      >
         Button
      </div>
    </div>
  </button>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: hsla(222, 63%, 34%, 1);
  background: hsla(222, 62%, 92%, 1);
  padding: 0.25rem 0.75rem;
  height: 2.25rem;
}

.emotion-2:hover {
  color: hsla(222, 63%, 34%, 1);
  background: hsla(222, 64%, 84%, 1);
}

.emotion-2:active {
  color: hsla(222, 63%, 34%, 1);
  background: hsla(222, 78%, 77%, 1);
}

.emotion-2:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

<button
    class="emotion-2"
    type="button"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-0"
      >
         Button
      </div>
    </div>
  </button>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: hsla(222, 63%, 43%, 1);
  background: transparent;
  padding: 0.25rem 0.75rem;
  height: 2.25rem;
}

.emotion-2:hover {
  color: hsla(222, 63%, 43%, 1);
  background: hsla(224, 65%, 97%, 1);
}

.emotion-2:active {
  color: hsla(222, 63%, 43%, 1);
  background: hsla(222, 62%, 92%, 1);
}

.emotion-2:disabled {
  background: transparent;
  color: hsla(0, 0%, 43%, 1);
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

<button
    class="emotion-2"
    type="button"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-0"
      >
         Button
      </div>
    </div>
  </button>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #FFFFFF;
  background: hsla(0, 58%, 52%, 1);
  padding: 0.25rem 0.75rem;
  height: 2.25rem;
}

.emotion-2:hover {
  color: #FFFFFF;
  background: hsla(0, 58%, 37%, 1);
}

.emotion-2:active {
  color: #FFFFFF;
  background: hsla(0, 58%, 30%, 1);
}

.emotion-2:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 59%, 83%, 1);
}

<button
    class="emotion-2"
    type="button"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-0"
      >
         Button
      </div>
    </div>
  </button>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: hsla(0, 58%, 37%, 1);
  background: hsla(0, 60%, 93%, 1);
  padding: 0.25rem 0.75rem;
  height: 2.25rem;
}

.emotion-2:hover {
  color: hsla(0, 58%, 37%, 1);
  background: hsla(0, 59%, 83%, 1);
}

.emotion-2:active {
  color: hsla(0, 58%, 37%, 1);
  background: hsla(0, 81%, 77%, 1);
}

.emotion-2:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 59%, 83%, 1);
}

<button
    class="emotion-2"
    type="button"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-0"
      >
         Button
      </div>
    </div>
  </button>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: hsla(0, 58%, 52%, 1);
  background: transparent;
  padding: 0.25rem 0.75rem;
  height: 2.25rem;
}

.emotion-2:hover {
  color: hsla(0, 58%, 52%, 1);
  background: hsla(0, 53%, 97%, 1);
}

.emotion-2:active {
  color: hsla(0, 58%, 52%, 1);
  background: hsla(0, 60%, 93%, 1);
}

.emotion-2:disabled {
  background: transparent;
  color: hsla(0, 0%, 43%, 1);
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 59%, 83%, 1);
}

<button
    class="emotion-2"
    type="button"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-0"
      >
         Button
      </div>
    </div>
  </button>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #000000;
  background: transparent;
  padding: 0.25rem 0.75rem;
  height: 2.25rem;
}

.emotion-2:hover {
  color: #000000;
  background: rgba(0, 0, 0, 0.05);
}

.emotion-2:active {
  color: #000000;
  background: rgba(0, 0, 0, 0.1);
}

.emotion-2:disabled {
  background: transparent;
  color: hsla(0, 0%, 43%, 1);
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(0, 0%, 87%, 1);
}

<button
    class="emotion-2"
    type="button"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-0"
      >
         Button
      </div>
    </div>
  </button>
  @keyframes animation-0 {
  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes animation-1 {
  0% {
    stroke-dasharray: 1,150;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 90,150;
    stroke-dashoffset: -35;
  }

  100% {
    stroke-dasharray: 90,150;
    stroke-dashoffset: -124;
  }
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #FFFFFF;
  background: hsla(222, 63%, 43%, 1);
  padding: 0.25rem 0.75rem;
  height: 2.25rem;
}

.emotion-5:hover {
  color: #FFFFFF;
  background: hsla(222, 63%, 34%, 1);
}

.emotion-5:active {
  color: #FFFFFF;
  background: hsla(222, 63%, 25%, 1);
}

.emotion-5:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-5:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-5:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  visibility: hidden;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  position: absolute;
  bottom: 0rem;
  top: 0rem;
  left: 0rem;
  right: 0rem;
}

.emotion-2 {
  -webkit-animation: animation-0 1.5s linear infinite;
  animation: animation-0 1.5s linear infinite;
  width: 24px;
  height: 24px;
}

.emotion-1 {
  fill: none;
  stroke: currentColor;
  stroke-width: 5;
  stroke-linecap: round;
  -webkit-animation: animation-1 1s ease-in-out infinite;
  animation: animation-1 1s ease-in-out infinite;
}

<button
    aria-disabled="true"
    class="emotion-5"
    disabled=""
    style="pointer-events: none;"
    type="button"
  >
    <div
      class="emotion-4"
    >
      <div
        class="emotion-0"
      >
         Button
      </div>
      <div
        class="emotion-3"
      >
        <svg
          class="emotion-2"
          focusable="false"
          viewBox="0 0 50 50"
        >
          <circle
            class="emotion-1"
            cx="25"
            cy="25"
            r="20"
          />
        </svg>
      </div>
    </div>
  </button>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #FFFFFF;
  background: hsla(222, 63%, 43%, 1);
  padding: 0.25rem 0.75rem;
  height: 2.25rem;
}

.emotion-2:hover {
  color: #FFFFFF;
  background: hsla(222, 63%, 34%, 1);
}

.emotion-2:active {
  color: #FFFFFF;
  background: hsla(222, 63%, 25%, 1);
}

.emotion-2:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

<button
    aria-disabled="true"
    class="emotion-2"
    disabled=""
    style="pointer-events: none;"
    type="button"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-0"
      >
         Button
      </div>
    </div>
  </button>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #FFFFFF;
  background: hsla(222, 63%, 43%, 1);
  padding: 0.25rem 0.75rem;
  height: 2.25rem;
  margin-top: -0.5rem;
  margin-bottom: -0.5rem;
}

.emotion-2:hover {
  color: #FFFFFF;
  background: hsla(222, 63%, 34%, 1);
}

.emotion-2:active {
  color: #FFFFFF;
  background: hsla(222, 63%, 25%, 1);
}

.emotion-2:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

<button
    class="emotion-2"
    type="button"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-0"
      >
         Button
      </div>
    </div>
  </button>
  .emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 600;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.0125em;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  color: #FFFFFF;
  background: hsla(222, 63%, 43%, 1);
  padding: 0.25rem 0.75rem;
  height: 2.25rem;
  margin-left: -0.75rem;
  margin-right: -0.75rem;
}

.emotion-2:hover {
  color: #FFFFFF;
  background: hsla(222, 63%, 34%, 1);
}

.emotion-2:active {
  color: #FFFFFF;
  background: hsla(222, 63%, 25%, 1);
}

.emotion-2:disabled {
  background: hsla(0, 0%, 87%, 1);
  color: hsla(0, 0%, 43%, 1);
}

.emotion-2:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-2:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem hsla(222, 64%, 84%, 1);
}

<button
    class="emotion-2"
    type="button"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-0"
      >
         Button
      </div>
    </div>
  </button>
</DocumentFragment>
`;
